pandas/core/series.py

```diff
@@ -700,6 +701,72 @@ def view(self, dtype=None):
 
     # ----------------------------------------------------------------------
     # NDArray Compat
+    _HANDLED_TYPES = (Index, ExtensionArray, np.ndarray, numbers.Number,
+                      list)  # what other builtins? array? deque? ...
```

Comment:
Iterable/Sequence?

Reply:
I'm not sure whether including ABCs here is appropriate or not. That kind of puts us in the highest position, even if we may want to raise (e.g. `np.add(Series, Xarray.DataSet)`). I don't know what's best hear.

Reply:
Instead of using `numbers.Number`, you could explicitly check for a lack of `__array_priority__` and `__array_ufunc__` attributes to indicate that something is a scalar / can be safely coerced.  That is basically what NumPy does.

Reply:
@jorisvandenbossche and I were discussing using `__array_priority__` the other day.  This seems less likely to accidentally leave things out than _HANDLED_TYPES.